cmd/kpod/stats.go

```diff
@@ -0,0 +1,181 @@
+	}
+	format := c.String("format")
+	all := c.Bool("all")
+	config, err := getConfig(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not read config")
+	}
+	containerServer, err := libkpod.New(config)
+	if err != nil {
+		return errors.Wrapf(err, "could not create container server")
+	}
+
+	containerIDs, err := getContainerList(containerServer, c.Args(), all)
+	if err != nil {
+		return errors.Wrapf(err, "could not get list of containers")
+	}
+	if c.Bool("no-stream") {
+		// print out statistics and be on your way
+		stats, err := containerServer.GetContainerStatsOnce(containerIDs)
+		if err != nil {
+			return errors.Wrapf(err, "could not get stats once")
+		}
+		outputStats(stats, format)
+		return nil
+	}
+	statsChan := make(chan []*libkpod.ContainerStats)
+	// iterate over the channel until it is closed
+	go func() {
+		for stats := range statsChan {
+			// Continually refresh statistics
+			tm.MoveCursor(1, 1)
+
+			outputStats(stats, format)
+
+			tm.Flush()
+
+			time.Sleep(time.Second)
+		}
+	}()
+	return containerServer.GetContainerStatsFollow(containerIDs, statsChan)
+}
+
+func getContainerList(server *libkpod.ContainerServer, args []string, all bool) ([]string, error) {
+	containers := []*oci.Container{}
+	ids := []string{}
+	if len(args) > 0 {
+		for _, id := range args {
+			if server.HasContainer(id) {
+				containers = append(containers, server.GetContainer(id))
+			} else {
+				return nil, fmt.Errorf("no such container: ", id)
+			}
+		}
+	} else {
+		containers = server.ListContainers()
+	}
+	for _, container := range containers {
+		if !all && container.State().Status != "running" {
+			continue
+		}
+		ids = append(ids, container.ID())
+	}
+	return ids, nil
+}
+
+func outputStats(stats []*libkpod.ContainerStats, format string) error {
+	if format == "" {
+		outputStatsHeader()
+	}
+	var err error
+	for _, s := range stats {
+		if format == "" {
+			outputStatsUsingFormatString(s)
+		} else {
+			params := getStatsOutputParams(s)
+			err2 := outputStatsUsingTemplate(format, params)
+			errors.Wrapf(err, err2.Error())
+		}
+	}
+	return err
+}
+
+func outputStatsHeader() {
+	fmt.Printf("%-64s %-16s %-20s %-16s %-20s %-20s %s", "CONTAINER", "CPU %", "MEM USAGE / MEM LIMIT", "MEM %", "NET I/O", "BLOCK I/O", "PIDS")
+}
+
+func outputStatsUsingFormatString(stats *libkpod.ContainerStats) {
+	fmt.Printf("%-64s %-16s % %-20s %-16s % %-20s %-20s %s", stats.Container, stats.CPU, combineHumanValues(stats.MemUsage, stats.MemLimit), stats.MemPerc, combineHumanValues(stats.NetInput, stats.NetOutput), combineHumanValues(stats.BlockInput, stats.BlockOutput), stats.PIDs)
+}
+
+func combineHumanValues(a, b uint64) string {
+	return fmt.Sprintf("%s / %s", libkpodimage.FormattedSize(float64(a)), libkpodimage.FormattedSize(float64(b)))
+}
+
+func floatToPercentString(f float64) string {
+	return fmt.Sprintf("%.2f %", f)
+}
+
+func getStatsOutputParams(stats *libkpod.ContainerStats) statsOutputParams {
```

Comment:
I left this out (and did the same thing with kpod images) because it directly relates to output, which I thought should be client-specific.  If you think it should be moved though I can do that